src/transformers/models/speecht5/feature_extraction_speecht5.py

```diff
@@ -160,46 +160,24 @@ def zero_mean_unit_var_norm(
 
         return normed_input_values
 
-    @staticmethod
-    def _stft(waveform: np.ndarray, fft_length: int, hop_length: int, window: np.ndarray) -> np.ndarray:
-        """
-        Calculates the magnitude spectrogram over one waveform array.
-        """
-        # center pad the waveform
-        padding = [(int(fft_length // 2), int(fft_length // 2))]
-        waveform = np.pad(waveform, padding, mode="reflect")
-        waveform_size = waveform.size
-
-        # promote to float64, since np.fft uses float64 internally
-        waveform = waveform.astype(np.float64)
-
-        num_frames = int(1 + np.floor((waveform_size - fft_length) / hop_length))
-        num_frequency_bins = (fft_length // 2) + 1
-        spectrogram = np.empty((num_frames, num_frequency_bins))
-
-        start = 0
-        for frame_idx in range(num_frames):
-            frame = waveform[start : start + fft_length] * window
-            spectrogram[frame_idx] = np.abs(np.fft.rfft(frame))
-            start += hop_length
-
-        return spectrogram
-
     def _extract_mel_features(
         self,
         one_waveform: np.ndarray,
     ) -> np.ndarray:
         """
         Extracts log-mel filterbank features for one waveform array (unbatched).
         """
-        if self.n_fft != self.sample_size:
-            raise NotImplementedError(
-                f"Currently the STFT frame size must be a power of two, but got {self.sample_size} for a window length of {self.win_length} and sampling rate of {self.sampling_rate}. Ensure `win_length * sampling_rate // 1000` is divisible by two."
-            )
-
-        stft_out = self._stft(one_waveform, self.n_fft, self.sample_stride, self.window)
-
-        return np.log10(np.maximum(self.mel_floor, np.dot(stft_out, self.mel_filters)))
+        log_mel_spec = spectrogram(
+            one_waveform,
+            window=self.window,
+            frame_length=self.sample_size,
+            hop_length=self.sample_stride,
+            fft_length=self.n_fft,
+            mel_filters=self.mel_filters,
+            mel_floor=self.mel_floor,
+            log_mel="log10",
+        )
+        return log_mel_spec.T
```

Comment:
(nit) Is it clearer to transpose here or in the `__call__`? Just wonder whether we can have consistency across our feature extractors by always returning the non-transposed version?

Reply:
I've used the shapes as returned by librosa and torchaudio, which is `(bins, length)` but not every model uses it in that order. For existing models I didn't want to change the shape of tensors returned by the feature extractors.

To me it makes most sense to have the feature extractor return the spectrogram in the shape the model intends to use it. 

(BTW, right now none of the feature extractors actually documents its return values.)

Reply:
Sounds good to me!